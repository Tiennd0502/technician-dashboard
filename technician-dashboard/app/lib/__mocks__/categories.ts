export const CATEGORIES = [
  {
    id: '01',
    label: 'Sofa Cleaning',
    image: 'https://i.ibb.co/QcSmTHW/category01.webp',
  },
  {
    id: '02',
    label: 'Bathroom',
    image: 'https://i.ibb.co/KDQzZ3s/category02.web',
  },
  {
    id: '03',
    label: 'Best Control',
    image: 'https://i.ibb.co/jwsPTTt/category03.webp',
  },
  {
    id: '04',
    label: 'View All',
    image: 'https://i.ibb.co/KxsM943/category04.webp',
  },
];
