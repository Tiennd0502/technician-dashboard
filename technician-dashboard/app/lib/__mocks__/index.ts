export * from './table';
export * from './categories';
export * from './services';
