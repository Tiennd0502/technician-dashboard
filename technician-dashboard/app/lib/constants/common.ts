export const LOGO_URL = '/images/technician.svg';

export const AVATAR_DEFAULT = '/images/empty-avatar.svg';

export const CURRENCY_CHARACTERS = {
  EURO: 'â‚¬',
};

export const TRUNCATE_STYLE = {
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
};

export const NO_DATA = 'No data found';
