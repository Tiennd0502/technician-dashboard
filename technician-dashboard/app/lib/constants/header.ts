import { AVATAR_DEFAULT } from '.';

export const HEADER_INFO = {
  user: {
    name: 'Belle',
    avatar: AVATAR_DEFAULT,
  },
  title: 'Product & Services',
};
