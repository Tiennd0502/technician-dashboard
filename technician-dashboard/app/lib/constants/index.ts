export * from './common';
export * from './filter';
export * from './sidebar';
export * from './header';
export * from './rules';
export * from './messages';
export * from './routes';
export * from './status';
export * from './table';
export * from './categories';
export * from './services';
