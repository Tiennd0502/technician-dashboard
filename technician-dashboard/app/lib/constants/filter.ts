import { SORT_TYPE } from '@/lib/interfaces';

export const PRODUCT_LIMIT = 5;

export const DEFAULT_PRODUCT_FILTER = {
  name: '',
  sortBy: 'name',
  order: SORT_TYPE.Asc,
};
