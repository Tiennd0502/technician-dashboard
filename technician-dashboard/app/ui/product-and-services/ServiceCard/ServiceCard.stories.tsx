import type { Meta, StoryObj } from '@storybook/react';

import { SERVICES } from '@/lib/__mocks__';
import { CURRENCY_CHARACTERS } from '@/lib/constants';

import ServiceCard from '.';

const meta: Meta<typeof ServiceCard> = {
  title: 'Components/ServiceCard',
  component: ServiceCard,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof ServiceCard>;

export const Default: Story = {
  args: { service: SERVICES[0] },
};

export const HasCurrencyCharacter: Story = {
  args: { service: SERVICES[0], currencyCharacter: CURRENCY_CHARACTERS.EURO },
};
