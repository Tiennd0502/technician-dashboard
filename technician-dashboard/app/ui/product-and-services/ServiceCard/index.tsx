import { memo } from 'react';
import { Link, Spacer, Text, Flex } from '@chakra-ui/react';

import { TRUNCATE_STYLE } from '@/lib/constants';
import { Service } from '@/lib/interfaces';

interface ServiceCardProps {
  service: Service;
  currencyCharacter?: string;
}

const ServiceCard = ({
  service: { id, name, image, price },
  currencyCharacter = '',
}: ServiceCardProps) => (
  <Link
    key={id}
    minH={{ base: '100px', sm: '120px', md: '165px' }}
    w='100%'
    href='#'
    borderRadius='md'
    overflow='hidden'
    _hover={{
      opacity: '0.7',
    }}
    bg={`url('${image}') center center no-repeat`}
    bgSize='cover'
  >
    <Flex flexDirection='column' justify-content='flex-end' h='100%'>
      <Spacer />
      <Text
        color='text.tertiary'
        variant={{ base: 'textXs', md: 'textMd' }}
        px='2.5'
        {...TRUNCATE_STYLE}
      >
        {name}
      </Text>
      <Text
        color='text.tertiary'
        variant={{ base: 'textXs', md: 'textMd' }}
        px='2.5'
        mb='3'
      >{`${currencyCharacter}${price}`}</Text>
    </Flex>
  </Link>
);

export default memo(ServiceCard);
