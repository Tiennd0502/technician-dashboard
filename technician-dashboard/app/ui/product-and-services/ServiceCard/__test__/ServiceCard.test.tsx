import { render } from '@testing-library/react';

import { SERVICE } from '@/lib/__mocks__';
import { CURRENCY_CHARACTERS } from '@/lib/constants';

import ServiceCard from '..';

describe('ServiceCard test cases', () => {
  const props = {
    service: SERVICE,
    currencyCharacter: CURRENCY_CHARACTERS.EURO,
  };

  test('should render correctly', () => {
    const { container } = render(<ServiceCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
