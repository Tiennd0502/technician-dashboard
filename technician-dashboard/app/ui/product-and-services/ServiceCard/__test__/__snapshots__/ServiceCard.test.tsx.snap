// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard test cases should render correctly 1`] = `
<div>
  <a
    class="chakra-link css-qwh07e"
    href="#"
  >
    <div
      class="css-1g4yje1"
      justify-content="flex-end"
    >
      <div
        class="css-17xejub"
      />
      <p
        class="chakra-text css-od0eia"
      >
        Carpenter
      </p>
      <p
        class="chakra-text css-18bew60"
      >
        â‚¬45
      </p>
    </div>
  </a>
</div>
`;
